capstone/src/main/resources/static/user.js

```diff
@@ -0,0 +1,375 @@
+  const form = document.createElement("form");
+  form.setAttribute("method", "POST");
+  form.setAttribute("action", "add-saved-organization?organization-id=" + data.datastoreId);
+  const button = document.createElement('button');
+  button.innerText = "Save this organization";
+  button.setAttribute("type", "submit");
+
+  form.appendChild(button);
+  return form;
+}
+
+/* create the individual event containers for displaying events*/
+function createSavedEventElement(event, saveAllowed) {
+  const divElement = document.createElement('div');
+  divElement.setAttribute("class", "item-container general-container");
+
+  const h3ElementName = document.createElement('h3');
+  h3ElementName.innerText = event.eventTitle;
+  divElement.appendChild(h3ElementName);
+
+  const pElementTime = document.createElement('p');
+  pElementTime.innerText = event.eventDateTime;
+  divElement.appendChild(pElementTime);
+
+  // create delete event form
+  const form = saveAllowed? createUnsaveEventButton(event) : createEditEventButton(event);
+  divElement.appendChild(form);
+
+  return divElement;
+}
+
+/* creates an unsave button for event */
+function createUnsaveEventButton(event) {
+  const form = document.createElement("form");
+  form.setAttribute("method", "POST");
+  form.setAttribute("action", "delete-saved-event?event-id=" + event.datastoreID);
+  const button = document.createElement('button');
+  button.innerText = "Unsave this event";
+  button.setAttribute("type", "submit");
+
+  form.appendChild(button);
+  return form;
+}
+
+/* creates an edit button for events */
+function createEditEventButton(event) {
+  const form = document.createElement("form");
+  form.setAttribute("action", "event.html#" + event.datastoreID);
+  const button = document.createElement('button');
+  button.innerText = "Edit this event";
+  button.setAttribute("type", "submit");
+
+  form.appendChild(button);
+  return form;
+}
+
+/* Function to control form display using button */
+function revealForm() {
+	var userType = sessionStorage.getItem("user-type");
+	if (userType == null) {
+		getUserType();
+      if (sessionStorage.getItem("user-type") == null) {
+        displayForm("individual", true);
+        return;
+      }
+    }
+  displayForm(userType, false);
+}
+
+/* Function to close form display after submission */
+function closeForm() {
+	document.getElementById("user").style.display = "none";
+	document.getElementById("organization").style.display = "none";
+}
+
+/* function to get all events hosted by the current organization */
+function getOrganizationEvents() {
+  var userType = sessionStorage.getItem("user-type");
+  if (userType == null) {
+    getUserType();
+  }
+  if (userType == "organization") {
+    fetch('get-' + userType).then(response => response.json()).then((data) => {
+      const eventDiv = document.getElementById("created-events");
+      data[0].events.forEach((event) => eventDiv.appendChild(createSavedEventElement(event), false));
```

Comment:
Ok I did a super quick poke around and I think the error is that there the events list is null for most orgs

Reply:
Yeah I just realized that most of the test data for organizations are out of date and do not have the required fields. I will try to add some on Monday.

Reply:
Sorry I don't want to leave you hanging with the error. I'm not sure if you used an old account instead of creating a new one when testing, but I think you got the error because some of the fields for organizations changed types. The error would resolve itself if you delete the current account and create a new one.

Reply:
works!